File: app/src/main/java/com/fj/temanteman/database/FriendHelper.java
package com.fj.temanteman.database;

import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.database.SQLException;
import android.database.sqlite.SQLiteDatabase;

import com.fj.temanteman.models.Friend;

import java.util.ArrayList;

import static android.provider.BaseColumns._ID;
import static com.fj.temanteman.database.DatabaseContract.FriendColumns.INITIAL;
import static com.fj.temanteman.database.DatabaseContract.FriendColumns.MAJOR;
import static com.fj.temanteman.database.DatabaseContract.FriendColumns.NAME;
import static com.fj.temanteman.database.DatabaseContract.FriendColumns.PHONE;
import static com.fj.temanteman.database.DatabaseContract.TABLE_FRIEND;

public class FriendHelper {

    private static String DATABASE_TABLE = TABLE_FRIEND;
    private Context context;
    private DatabaseHelper dataBaseHelper;

    private SQLiteDatabase database;

    public FriendHelper(Context context){
        this.context = context;
    }

    public void open() throws SQLException {
        dataBaseHelper = new DatabaseHelper(context);
        database = dataBaseHelper.getWritableDatabase();
    }

    public void close(){
        dataBaseHelper.close();
    }

    public ArrayList<Friend> query(){
        ArrayList<Friend> arrayList = new ArrayList<>();
        Cursor cursor = database.query(DATABASE_TABLE,
                null,null,null,null,null,_ID +" DESC",null);
        cursor.moveToFirst();
        Friend friend;

        if (cursor.getCount() > 0) {
            do {
                friend = new Friend();
                friend.setId(cursor.getInt(cursor.getColumnIndexOrThrow(_ID)));
                friend.setInitial(cursor.getString(cursor.getColumnIndexOrThrow(INITIAL)));
                friend.setName(cursor.getString(cursor.getColumnIndexOrThrow(NAME)));
                friend.setMajor(cursor.getString(cursor.getColumnIndexOrThrow(MAJOR)));
                friend.setPhone(cursor.getString(cursor.getColumnIndexOrThrow(PHONE)));

                arrayList.add(friend);
                cursor.moveToNext();
            } while (!cursor.isAfterLast());
        }

        cursor.close();
        return arrayList;
    }

    public void insert(Friend friend){
        ContentValues initialValues =  new ContentValues();
        initialValues.put(INITIAL, friend.getInitial());
        initialValues.put(NAME, friend.getName());
        initialValues.put(MAJOR, friend.getMajor());
        initialValues.put(PHONE, friend.getPhone());
        database.insert(DATABASE_TABLE, null, initialValues);
    }

    public void delete(int id){
        database.delete(TABLE_FRIEND, _ID + " = '" + id + "'", null);
    }
}

File: app/src/main/java/com/fj/temanteman/database/FriendHelper.kt
package com.fj.temanteman.database

import android.content.ContentValues
import android.content.Context
import android.database.SQLException
import android.database.sqlite.SQLiteDatabase

import com.fj.temanteman.models.Friend

import java.util.ArrayList

import com.fj.temanteman.database.DatabaseContract.FriendColumns.Companion.ID
import com.fj.temanteman.database.DatabaseContract.FriendColumns.Companion.INITIAL
import com.fj.temanteman.database.DatabaseContract.FriendColumns.Companion.MAJOR
import com.fj.temanteman.database.DatabaseContract.FriendColumns.Companion.NAME
import com.fj.temanteman.database.DatabaseContract.FriendColumns.Companion.PHONE
import com.fj.temanteman.database.DatabaseContract.TABLE_FRIEND
import kotlin.jvm.Throws

class FriendHelper(private val context: Context) {

    private val table = TABLE_FRIEND

    private lateinit var dataBaseHelper: DatabaseHelper
    private lateinit var database: SQLiteDatabase

    @Throws(SQLException::class)
    fun open() {
        dataBaseHelper = DatabaseHelper(context)
        database = dataBaseHelper.writableDatabase
    }

    fun close() {
        dataBaseHelper.close()
    }

    fun query(): ArrayList<Friend> {
        val arrayList = ArrayList<Friend>()
        val cursor = database.query(table, null, null, null, null, null, "id DESC", null)
        cursor.moveToFirst()
        var friend: Friend

        if (cursor.count > 0) {
            do {
                friend = Friend()
                friend.id = cursor.getInt(cursor.getColumnIndexOrThrow(ID))
                friend.initial = cursor.getString(cursor.getColumnIndexOrThrow(INITIAL))
                friend.name = cursor.getString(cursor.getColumnIndexOrThrow(NAME))
                friend.major = cursor.getString(cursor.getColumnIndexOrThrow(MAJOR))
                friend.phone = cursor.getString(cursor.getColumnIndexOrThrow(PHONE))

                arrayList.add(friend)
                cursor.moveToNext()
            } while (!cursor.isAfterLast)
        }

        cursor.close()
        return arrayList
    }

    fun insert(friend: Friend) {
        val initialValues = ContentValues()
        initialValues.put(INITIAL, friend.initial)
        initialValues.put(NAME, friend.name)
        initialValues.put(MAJOR, friend.major)
        initialValues.put(PHONE, friend.phone)
        database.insert(table, null, initialValues)
    }

    fun delete(id: Int) {
        database.delete(table, "id = '$id'", null)
    }
}
